CONTRIBUTING_GUIDELINES.md

```diff
@@ -0,0 +1,287 @@
+Branch names should be in the format **\<issue-number\>-kebab-case-title**
+
+All branches should have distinct history and should be visually easy to follow, for this reason only perform merge commits when merging code either by PR or when synchronising.
+
+If you wish to rebase you should be following the [Golden Rule](https://www.atlassian.com/git/tutorials/merging-vs-rebasing#the-golden-rule-of-rebasing) and ahere to the advice in the heading [Aside: Rebase as cleanup is awesome in the coding lifecycle](https://www.atlassian.com/git/articles/git-team-workflows-merge-or-rebase).
+
+## Step:three: Work Remotely 👩‍💻
+- Now you are ready to your work to the remote repository.
+- Perfom your desired changes to the code base.
+- Add all the files/folders needed.  
+- Check the outcome
+- Make a small clip or take screenshots
+
+<p align="center"><img width=35% src="https://media2.giphy.com/media/L1R1tvI9svkIWwpVYr/giphy.gif?cid=ecf05e47pzi2rpig0vc8pjusra8hiai1b91zgiywvbubu9vu&rid=giphy.gif"></p>
+
+
+##### 1. To add all changes :clipboard:
+
+- After you've made changes or made your contribution to the project add changes to the branch you've just created by: 
+
+###### To add all new files to branch Branch_Name 
+
+```   
+$ git add .  
+```
+
+###### To add only a few files to Branch_Name
+
+```
+$ git add <some_file_name>
+```
+##### 2. Commit your changes :pencil2:
+
+- To commit give a descriptive message for the convenience of reviewer.
+- This message get associated with all files you have changed. 
+
+```
+git commit -m "Relevant message"
+```
+
+##### 3. Push the committed changes in your feature branch to your remote repo.
+
+```
+git push -u origin <your_branch_name>
+```
+## step:four: PR Guidlines
+
+ - To create a pull request, click on `compare and pull requests`. Please ensure you compare your feature branch to the desired branch of the repo you are suppose to make a PR to.
+
+
+
+##### 1. Then add an appropriate title and description to your pull request that explains your changes and efforts done.
+
+
+##### 2. Click on `Create Pull Request`
+ 
+  ![pullrequest-send](readme_assets/pullrequest-send.jpg)  
+  
+![pullrequest-description](readme_assets/pullrequest-description.png)
+- Voila! You have made a PR to the doc2pen project boom . Sit back patiently and relax while the project maintainers review your PR and merged.🥳 
+ 
+## Linting :grey_exclamation:
+
+PR should be linted properly locally. There is no system restriction applied for this, however, PRs will not be merged until they contain properly formatted code.
+
+## Continous Integration
+
+All PRs will have all these tests run and a combined coverage report will be attached, if coverage is to go down the PR will be marked failed. On Travis CI the automated tests are run on an emulator.
+
+## Need more help?🤔  
```

Comment:
What is the point of having this when we are making a dedicated guide for the same thing